### ISSUE TITLE
Docker For Windows Beta 27.1 networking issue (HNS failed with error)

### ISSUE THREAD

**Issue by mpwagne** [docker/for-win#123]
**Issue**
The new Docker For Windows Beta 27.1 is causing an issue for my windows containers (this was working fine on Beta 27 last week).  Seems to be some kind of networking issue.

**When executing run command:**
docker run -e environment=LOCAL -d -p 80:80 --name tc1 testapi

**I get this error:**
C:\Program Files\Docker\Docker\Resources\bin\docker.exe: Error response from daemon: failed to create endpoint tc1 on network nat: HNS failed with error : Unspecified error.

I have uninstalled DfW and cleared the docker folders (programdata and programfiles) then reinstalled - this did not help, get the same issue. 
Clue(?): If I take the port param out of the command it seems to generate the container without error.

### Repro:
**(Eliminate my images)**
docker run -d -p 80:80 --name tc2 microsoft/iis
this gives same error...

### Information
Diagnostic ID: 8CC04A2F-A1F5-4858-B417-CB8B725532F8/2016-10-03_10-35-55

-------
**docker version**
Client:
 Version:      1.12.2-rc1
 API version:  1.24
 Go version:   go1.6.3
 Git commit:   45bed2c
 Built:        Tue Sep 27 23:38:15 2016
 OS/Arch:      windows/amd64
 Experimental: true

Server:
 Version:      1.12.2-cs2-ws-beta-rc1
 API version:  1.25
 Go version:   go1.7.1
 Git commit:   62d9ff9
 Built:        Fri Sep 23 20:50:29 2016
 OS/Arch:      windows/amd64

-------

**docker info**
Containers: 1
 Running: 0
 Paused: 0
 Stopped: 1
Images: 11
Server Version: 1.12.2-cs2-ws-beta-rc1
Storage Driver: windowsfilter
 Windows:
Logging Driver: json-file
Plugins:
 Volume: local
 Network: nat null overlay transparent
Swarm: inactive
Security Options:
Kernel Version: 10.0 14393 (14393.206.amd64fre.rs1_release.160915-0644)
Operating System: Windows 10 Enterprise
OSType: windows
Architecture: x86_64
CPUs: 8
Total Memory: 15.91 GiB
Name: MYWIN10
ID: WZRF:MHEG:3ELD:BF4K:2VCJ:S6HG:5BT6:P4DQ:TEXP:K6AC:OCLD:RGMT
Docker Root Dir: C:\ProgramData\Docker
Debug Mode (client): false
Debug Mode (server): false
Username: myuser
Registry: https://index.docker.io/v1/
Insecure Registries:
 127.0.0.0/8

**Comment by friism** [comment]
Can you check whether something else might be running on port 80? http://stackoverflow.com/a/48199/2942

Also, did you clean out all of `$env:programdata\docker`?

**Comment by mpwagne** [comment]
Thanks for prompt response.

I checked resmon and Port 80 is free.  To double confirm I also tried:
docker run -d -p 83:80 --name tc2 microsoft/iis
received same error.

Yes, to clarify my cleaning steps... 
- Uninstalled Docker for Windows
- Deleted C:\Program Files\Docker
- Deleted C:\ProgramData\Docker  (after taking ownership)
- Deleted MobyLinuxVM vm 
- Reinstalled Docker for Windows/ shifted into windows container mode

**NOTE:** 
I tried the cleaning steps again but this time I installed the original Beta 26 installer I had loaded and ignored upgrade warnings. **Note that I get the same error with the old DfW version (beta 26).** Makes me question if there might be something network related that I'm not fully cleaning...?  I don't think I ran beta 26 originally - I used the installer but jumped to beta 27 right away using upgrade (to beta 27 not 27.1). beta 27 was working fine on Friday afternoon. This morning I fired up the DfW UI and it strongly warned me to upgrade to 27.1 so I happily agreed. That is where things fell apart for me.   :-)

Are there full cleanup steps somewhere (for windows)?
